### ISSUE TITLE
Are collection trees already implemented in Invenio 3?

### ISSUE THREAD

**Issue by fjorba** [inveniosoftware/troubleshooting#25]
From time to time, I peek into Inveniosoftware repos, and the examples of Invenio 3 that you gave in Garching, and I have been able to find any evidence that collection trees are implemented in Invenio 3.

As collection trees, real and virtual, are a fundamental functionality of Invenio, we wonder when are they going to be implemented.  Which are your plans for that?

Thanks

**Comment by jacquerie** [comment]
[Invenio-Collections](https://github.com/inveniosoftware/invenio-collections) implements collection trees. See for example: https://github.com/inveniosoftware/invenio-collections/blob/f3adca45c6d00a4dbf1f48fd501e8a68fe347f2f/tests/test_percolator.py#L77-L213.

(This repository is not meant for this kind of questions, but it doesn't matter.)

**Comment by fjorba** [comment]
Sorry if I misunderstood, I thought from Lars presentation in Garching that all questions would be welcome, so they can be searched and reused for the whole community.

Anyway, looking at the code I'm unable to evaluate how much of it is implemented, has a admin user interface and a public view.  Do you know of any Invenio 3 that has it running, even in test (labs) mode?

**Comment by jacquerie** [comment]
> Do you know of any Invenio 3 that has it running, even in test (labs) mode?

In INSPIRE we are using `invenio-collections`, but we are not using an admin user interface or any kind of view: we use its information to filter records at the API level. We also don't use/need nested collections.

> I thought from Lars presentation in Garching that all questions would be welcome

Since I wasn't there and you were there, I trust that you're right on that. Sorry for the comment : )

**Comment by lnielsen** [comment]
Yes, this repository is meant for this type of question as well :-) It's both for developers and users. Right now we have some few posts that it doesn't make sense to have two separate places. 

As for collection trees then no there's nothing there yet which have been tested out on production systems. Besides INSPIRE, I know CERN Open Data looked at it at some point, but I don't know the status - perhaps @tiborsimko can comment?

**Comment by tiborsimko** [comment]
The CERN Open Data on Invenio 3 was released last December.  We did not use the `invenio-collection` package at the end, because it wasn't really necessary. What we do is simply keep redirecting people from old `/collection/XXX` pages to new coresponding `/search?q=YYY` faceted search page destination.  This approach may be a good example if one wants to "soft-deprecate" the notion of browsable collections and drive the users instead towards the new rich faceted search interface.  (Basically, the Invenio 3 search experience is considerably improved and users can use the  new "interactive faceted search exerience" instead of the old "browse static collection tree pages" experience.) This works easily on CERN Open Data because we don't have any restricted collections etc.

**Comment by fjorba** [comment]
Understood.  But we need collections, and we need nested collections and also virtual collections.  That was one of the reasons we choose Invenio, after all.  Recently I discovered another Invenio 3 site: https://videos.cern.ch, also without collection trees.

Are you aware of any Invenio 3 site with collection trees like classical Invenio?  Can we count on it being implemented? When?

**Comment by lnielsen** [comment]
On Zenodo we have plans within the next year to add support for collection trees in the community module. The design is still to be done, but so far we're not planning to support e.g. virtual collections.

I'm not aware of others right now, that are doing or planning anything with collections. Hence, the answer today to your question is, no, you cannot count on it being implemented as v1 as nobody have assigned any effort to it. If you ask again in 6-12 months, the answer might be different.

**Comment by fjorba** [comment]
Thanks @lnielsen  for the clarity of your answer.

However, this serious.  As I wrote previously, one of the top reasons we choose Invenio was the collection trees and the virtual collections.  We cannot even think about migrating to Invenio 3 until they are implemented, and a some (semi) automatic migration path exist, including portalboxes.  Of course, we can write our export/import code, and contribute upstream, but they should be supported by Invenio 3.

Collections are the base for organizing records, giving permissions, formating output, creating custom forms, exporting statistics, documenting the content via portalboxes, etc.  To give you an idea of the numbers, in our three Invenios we have currently almost 900 collections (455 for https://ddd.uab.cat, 356 for https://traces.uab.cat and 88 for https://ifmuc.uab.cat), growing continuously.  As most of them have custom portalboxes in three languages, that means about more than 2,600 portalboxes, most of them with some illustrative picture.

We depend on them.  Your answer means that we cannot plan an Invenio 3 migration until a year from now.

**Comment by lnielsen** [comment]
@fjorba I fully understand collections a la v1 is very important for you, and not having this available in v3 means you cannot migrate. You're likely also to have further issues since the Invenio-MARC21 module needs a lot of work to be a production-ready MARC21 implementation for v3. 

I wish we could solve the problem for you and all other v1 users, but the fact is that everybody is resource constrained, and I cannot justify spending CERN resources on bigger features that are not needed by a service at CERN.

In order to make collection trees and a proper MARC21 model happen, I think external contributors are needed, and I'm all ears for suggestions on how to make this happen.

**Comment by fjorba** [comment]
As I see it, this issue concerns the whole Invenio community, specially those who, like us at UAB, depend on classical collection management, and probably should be tackled in the next users meeting.

Without knowing the details, I'd say we are more prepared to collaborate on the Marc21 issues (as it is more of detail) that the collection management, that concerns architectural and UI techniques we are not familiarised with.
